### ISSUE TITLE
Feature request: Focus with depth support

### ISSUE THREAD

**Issue by ashubham** [sverweij/dependency-cruiser#564]
### Context

When currently using `--focus` option. It only shows the direct dependencies and dependents. In practice, I have been using `focus` to figure culprits which break code splitting (and hence lazy loading).

Since, I want to follow the chain of dependents to a large size module which I want to lazy load a depth option will be helpful.

### Expected Behavior

`--focus-depth N`  when using `--focus`.

This makes the output contain upto `N` dependents/depedencies from the focused modules.

### Current Behavior

`--focus` only shows direct dependents/dependencies.

### Considered alternatives

Currently I have to recursively call focus manually on the chain, to figure out the root culprits which unnecessary effort.

**Comment by JamieMason** [comment]
I'm looking for this feature too @ashubham and in the meantime have made the below script which I'm using, maybe a modified version of it could be useful to you too. It's pretty rough but it does what I need.

[[GIST] Find nested dependencies or imports of a JavaScript Module](https://gist.github.com/JamieMason/536f8b11d4f5ccfeade8144e1e114cea)

**Comment by sverweij** [comment]
Hi @ashubham / @JamieMason would something like this get you set?

### get the dependencies of one module

This will get you all dependencies of the-file-you-want-to-investigate.ts
```sh
depcruise src/the-file-you-want-to-investigate.ts -T text
```

This will get you all dependencies of the-file-you-want-to-investigate.ts to a [max depth](https://github.com/sverweij/dependency-cruiser/blob/develop/doc/cli.md#--max-depth) of 3
```sh
depcruise src/the-file-you-want-to-investigate.ts --max-depth 3 -T text
```

(I've used the `text` reporter here for demonstration purposes but this works with other reporters, e.g. `dot` as well)

### get the dependents of a module

If you have a need to see what's going _to_ a module - there's no cli option for that a.t.m. (could be a future feature). It _is_ possible to set up a rule for it, though. That way you'll prevent future regressions as well.

Example:
In the _forbidden_ section of your .dependency-cruiser.js add something like this:

```javascript
{
  name: "your-rule-name-here",
  severity: "info",
  from: {
      // if you want that module reachable from somewhere, or want
      // to limit where it's reachable from use the pathNot and path
      // attributes respectively. Leaving them out gives you _all_ modules
      // that reach the 
  },
  to: {
    path: "src/my-large-module-that-should-not-be-reachable\\.js$",
    reachable: true,
  },
}

```


### Use:
```sh
depcruise src --config 
```

#### Output

As an example I've added a rule similar to the one above in dependency-cruiser's own config to track everything reaching `src/extract/index.js`

<details>

<summary>click to see snippet</summary>

```json
    {
      "name": "your-rule-name-here",
      "severity": "info",
      "from": {
        "pathNot": "test"
      },
      "to": {
        "path": "src/extract/index\\.js$",
        "reachable": true
      }
    },
```

</details>

The default output (the `err` reporter) shows you all modules reaching the module (directly and transitively) + the routes going there. In the `dot` output the offending modules (not the routes...) would show up with a blue outline.

```text
 info your-rule-name-here: src/main/index.js → src/extract/index.js
      src/main/index.js →
      src/extract/index.js
  info your-rule-name-here: src/cli/index.js → src/extract/index.js
      src/cli/index.js →
      src/main/index.js →
      src/extract/index.js
  info your-rule-name-here: src/cli/format.js → src/extract/index.js
      src/cli/format.js →
      src/main/index.js →
      src/extract/index.js
  info your-rule-name-here: src/cli/format-meta-info.js → src/extract/index.js
      src/cli/format-meta-info.js →
      src/main/index.js →
      src/extract/index.js
  info your-rule-name-here: bin/dependency-cruise.js → src/extract/index.js
      bin/dependency-cruise.js →
      src/cli/index.js →
      src/main/index.js →
      src/extract/index.js
  info your-rule-name-here: bin/depcruise-fmt.js → src/extract/index.js
      bin/depcruise-fmt.js →
      src/cli/format.js →
      src/main/index.js →
      src/extract/index.js
  info your-rule-name-here: bin/depcruise-baseline.js → src/extract/index.js
      bin/depcruise-baseline.js →
      src/cli/index.js →
      src/main/index.js →
      src/extract/index.js

✖ 7 dependency violations (0 errors, 0 warnings). 477 modules, 1169 dependencies cruised.
```

**Comment by ashubham** [comment]
Thanks for your response @sverweij.

I am interested in the `dependents` use case, as its valuable to visualize what might be "bringing in" a large dependency. Is there a way to get a svg output from the `rules` interface ?

**Comment by sverweij** [comment]
Hi @ashubham the next version of dependency-cruiser (expected to land 2022-07-18) will have:
- `--reaches` command line option - see [`--reaches`](https://github.com/sverweij/dependency-cruiser/blob/develop/doc/cli.md#--reaches-show-modules-and-their-transitive-dependents) for details. This one is already available in `dependency-cruiser@11.12.0-beta-2`
- and additional `--focus-depth` command line option; see the linked PR (#636) for details.
